File: 1. Two Sum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        result = []
        for i in range(len(nums)):
            if target-nums[i] in dic:
                result.append(dic[target-nums[i]])
                result.append(i)
                return result
            else:
                dic[nums[i]] = i

File: 1. Two Sum/twoSum.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = {0,0};
        Hashtable<Integer, Integer> my_dict = new Hashtable<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (my_dict.get(nums[i]) != null) {
                res[0] = my_dict.get(nums[i]);
                res[1] = i;
                return res;
            }
            else{   
                my_dict.put(target-nums[i], i);
            }
        }
        return res;
    }
}
